cogstat/cogstat_stat.py

```diff
@@ -699,6 +699,168 @@ def variable_pair_standard_effect_size(data, meas_lev, sample=True, normality=No
 
+def calc_se_r2(r2, k, n):
+    """Calculate standard error of R-squared estimate.
+
+    Parameters
+    ----------
+    r2 : float
+        R-squared or adjusted R-squared point estimate.
+    k : int or float
+        Number of regressors in the model.
+    n : int or float
+        Number of observations.
+
+    Based on: Cohen, J., Cohen, P., West, S.G., and Aiken, L.S. (2003). Applied Multiple Regression/Correlation
+    Analysis for the Behavioral Sciences (3rd edition). Mahwah, NJ: Lawrence Earlbaum Associates. pp. 88
+    See also: https://stats.stackexchange.com/questions/175026/formula-for-95-confidence-interval-for-r2
+
+    Returns
+    -------
+    float
+    """
+
+    numerator = 4*r2*((1-r2)**2)*((n-k-1)**2)
+    denominator = ((n**2) - 1)*(3+n)
+    se_r2 = (numerator/denominator)**0.5
+    return se_r2
+
+def calc_r2_ci(r2, alpha, n, k):
+    """Calculate confidence interval of R-squared estimate.
+
+    Parameters
+    ----------
+    r2 : float
+        R-squared or adjusted R-squared point estimate.
+    alpha: float
+        Desired level of type-I error.
+    k : int or float
+        Number of regressors in the model.
+    n : int or float
+        Number of observations.
+
+    Based on: Olkin, I. and Finn, J.D. (1995). Correlations Redux. Psychological Bulletin, 118(1), pp. 155-164.
+    See also: https://www.danielsoper.com/statcalc/formulas.aspx?id=28
+    Validated against: https://www.danielsoper.com/statcalc/calculator.aspx?id=28 on 02/06/2022 06:03
+
+    Returns
+    -------
+    list
+    """
+
+    se_r2 = calc_se_r2(r2,k,n)
+    t = stats.t.ppf((1-alpha)/2, n-k-1)
+    up = r2 + t*se_r2
+    down = r2 - t*se_r2
+    return [up, down]
+
+def multiple_variables_standard_effect_size(data, x, y, normality, homoscedasticity, multicollinearity, sample=True,\
+                                            result=None):
+    """Calculate standardized effect size measures for multiple regression.
+
+    Parameters
+    ----------
+    data : pandas dataframe
+    x : list of str
+        Name of the explanatory variables.
+    y : list of str
+        Name of the dependent variable.
+    normality: bool or None
+        True if variables follow a multivariate normal distribution, False otherwise. None if normality couldn't be
+        calculated or if the parameter was not specified.
+    homoscedasticity : bool or None
+        True if variables are homoscedastic, False otherwise. None if homoscedasticity couldn't be calculated or
+        if the parameter was not specified.
+    multicollinearity:
+        True if possible multicollinearity was detected (VIF>10). None if the parameter was not specified.
+    sample : bool
+        True for sample descriptives, False for population estimations.
+    result : statsmodels regression result object
+        The result of the multiple regression analyses.
+
+    Returns
+    -------
+    html text
+    """
+    # TODO validate
+    import pingouin as pg
```

Comment:
it's already immported

please, do not abbreviate

Reply:
True, it missed my attention. Fixed.